iam.tf

```
@@ -14,30 +14,48 @@ locals {
 
 data "aws_partition" "current" {}
 
+data "aws_iam_role" "custom" {
+  count = var.create_iam_role ? 0 : 1
+  name  = var.custom_iam_role_name
+}
+
 resource "aws_iam_role" "this" {
   count = var.create_iam_role ? 1 : 0
   name  = local.base_name
   path  = "/"
 
   assume_role_policy = jsonencode({
     Version = "2012-10-17"
-    Statement = [{
-      Effect    = "Allow"
-      Action    = "sts:AssumeRole"
-      Principal = { Service = "ec2.${data.aws_partition.current.dns_suffix}" }
-    }]
+    Statement = concat([
+      {
+        Effect    = "Allow"
+        Action    = "sts:AssumeRole"
+        Principal = { Service = "ec2.${data.aws_partition.current.dns_suffix}" }
```

I know I added this code but this is wrong. even in govcloud, this'll be ec2.amazonaws.com.

```suggestion
        Principal = { Service = "ec2.amazonaws.com" }
```